controller/backing_image_controller.go

```diff
@@ -287,30 +283,27 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
 	return nil
 }
 
-func (bic *BackingImageController) IsBackingImageDataSourceCleanup(bi *longhorn.BackingImage) (finished bool, err error) {
+func (bic *BackingImageController) IsBackingImageDataSourceCleaned(bi *longhorn.BackingImage) (cleaned bool, err error) {
 	bids, err := bic.ds.GetBackingImageDataSource(bi.Name)
-	if err != nil && !apierrors.IsNotFound(err) {
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// can not get backing image data source, return cleaned so the controller won't wait until bids become failed-and-cleanup
```

Comment:
Actually you can use log trace for this comment. Ideally we should convert all comments to trace logs instead. (this can be improved later when you are working on log & error refactoring)